### PR TITLE
Add vnext contract

### PR DESCRIPTION
**Pull request by shenglol**

### CODE REVIEW

File: docs/vnext/contract.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+The maximum size of a request body that ARM will accept is 4 MB. Any request with a body larger than 4 MB will not be sent to the extensibility host and providers, and a 413 Payload Too Large will be returned to the client.
+
+### Maximum Response Size
+
+To keep it consistent with ARM's limit, the extensibility host does not accept any single resource response that exceeds 20MB in size. When this happens, the response body will be dropped and a 502 Bad Gateway error will be returned to the deployments RP.
+
+## Resource API Reference
+
+For details about resource APIs, Refer to the OpenAPI 3.0 based resource API [specification](openapi.yaml). You may view it in the ReDoc UI [here](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/Azure/bicep-extensibility/master/docs/vnext/openapi.yaml) for better experience.
```
Comment (shenglol) [inline docs/vnext/contract.md:34]: View the OpenAPI spec on the PR branch: https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/Azure/bicep-extensibility/shenglol/vnext-spec/docs/vnext/openapi.yaml&nocors

File: docs/vnext/openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+                $ref: '#/components/schemas/ResourceReference'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+  /{providerVersion}/resources/preview:
+    post:
+      summary: Create a preview of a resource create or update operation
+      description: |
+        Preview a resource creation or update operation by simulating the process.
+
+        Upon successful simulation, the outcome must mirror a resource that has been either created or updated, based on the operation that was previewed. This resulting state should align with what would be ontained from a GET response, as if the genuine create or update operation had been carried out. This includes the inclusion of read-only properties and properties endowed with default values, as well as the exclusion of write-only properties like those containing sensitive information such as passwords and connection strings.
```
Comment (shenglol) [inline docs/vnext/openapi.yaml:54]: The goal here is to let /preview support What-If.

File: docs/vnext/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    variables:
+      providerApiEndpoint:
+        default: 'api.example-provider.com'
+        description: Extensibility provider API endpoint.
+paths:
+  /{providerVersion}/resources/reference:
+    post:
+      summary: Create a resource reference
+      description: |
+        Generate a reference to a resource comprising a reference ID, regardless the existence of the resource.
```
Comment (shenglol) [inline docs/vnext/openapi.yaml:21]: The reference ID is used in /DELETE, which makes it possible for Deployment Stacks to manage extensible resources. We still need to finalize the design for authentication for the Kubernetes provider (see Anthony's proposal [here](https://gist.github.com/anthony-c-martin/9b7f8e8545c0848f3d8bfee234ba4360#part-1-authentication)), but for MS Graph it's not a problem.

File: docs/vnext/openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+        '200':
+          description: Successful operation
+          headers: 
+            Date:
+              $ref: '#/components/headers/Date'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResourceReference'
+        '400':
```
Comment (majastrz) [inline docs/vnext/openapi.yaml:53]: Technically many other HTTP status codes are possible to be returned (500, 503). Are we only including the intentional ones?

Comment (shenglol) [reply docs/vnext/openapi.yaml:53]: Good question. I can add a `default` response to cover all other cases.

File: docs/vnext/openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+          $ref: '#/components/headers/Date'
+      content: 
+        application/json: 
+          schema:
+            $ref: '#/components/schemas/ErrorResponse'
+    ResourceResponse:
+      description: Succesful operation
+      headers:
+        Date:
+          $ref: '#/components/headers/Date'
```
Comment (majastrz) [inline docs/vnext/openapi.yaml:277]: Should we support a version header in the response? (Could include it in our logs for investigations.)

**Review by majastrz — APPROVED** [review]

File: docs/vnext/openapi.yaml (hunk 5 above)
Comment (shenglol) [reply docs/vnext/openapi.yaml:310]: Good point. Added `x-ms-request-id` to make it consistent with ARM RPC.

**Pull request merged**